### ISSUE TITLE
tests/xml failing in CS

### ISSUE THREAD

**Issue by pmatos** [racket/racket#3011]
As of https://github.com/racket/racket/commit/848d2148b084e67ac6909cca4f1918271ee38207 we have a new CS failure:
https://github.com/racket/racket/runs/397792387#step:22:17

```
/home/runner/.racket/7.6.0.4/pkgs/racket-test/tests/xml/srcloc.rkt:16:1: test failure in (srcloc-position (first ((exn:srclocs-accessor exn) exn)))
  expected: 7
       got: (values #<thread-cell> #<procedure:...ate/more-scheme.rkt:261:28>)
  context...:
   do-error
   dynamic-wind
   call-with-values
   call-in-empty-metacontinuation-frame
   body of "/home/runner/.racket/7.6.0.4/pkgs/racket-test/tests/xml/srcloc.rkt"
   temp35_0
   for-loop
   run-module-instance!
   .../commands/test.rkt:77:2
   call-with-values
   call-in-empty-metacontinuation-frame
   body of (submod "/home/runner/.racket/7.6.0.4/pkgs/compiler-lib/compiler/commands/test.rkt" process)
   temp35_0
   body of top-level
   run-module-instance!
   eval-one-top
srcloc.rkt: raco test: non-zero exit: 1
```

**Comment by mflatt** [comment]
This appears to be a Chez Scheme bug (in cisco/ChezScheme, not just racket/ChezScheme) as illustrated by the program below. I will continue to investigate there.

```
;; cp0 is needed, probably to expose `record-ref`, but nothing looks
;; amiss in the cp0 output:
(run-cp0 (lambda (cp0 x)
           (let ([r (cp0 x)])
             #;(pretty-print (#%$uncprep r))
             r)))

(define miscompiled
  (lambda (bx)
    (define-record-type thing
      (fields pos)
      (nongenerative #{thing hlg584lmg5htbdauw7dkid2sh-0}))
    (set-box! bx (make-thing 'x))
    (let ([posx (unbox bx)])
      (cons 'result
            (call-with-values
                (lambda ()
                  ;; Works:
                  #;
                  (thing-pos posx)
                  ;; Does not work:
                  (if (thing? posx)
                      (thing-pos posx)
                      (do-something-else)))
              list)))))

;; Crashes instead of printing `(result x)`
(printf "~s\n" (miscompiled (box #f)))
```

**Comment by mflatt** [comment]
Addressed in cisco/ChezScheme#490, which is merged as racket/ChezScheme@d841744b26e4b4421fc4388b6943e558014e5e70.

**Comment by pmatos** [comment]
Thanks for the analysis and the fix here and in Chez.
